src/uucore/src/lib/features/format/num_format.rs

```diff
@@ -272,10 +278,32 @@ impl Formatter for Float {
             format_float_non_finite(x, self.case)
         };
 
+        // The format function will parse `x` together with its flag char,
+        // which should be placed in `flag`. So drop it here
+        s = if x < 0. { s[1..].to_string() } else { s };
+
+        let flag = get_flag(self.positive_sign, &x);
+
+        let remaining_width = if self.width >= flag.len() {
+            self.width - flag.len()
+        } else {
+            0
+        };
         match self.alignment {
-            NumberAlignment::Left => write!(writer, "{s:<width$}", width = self.width),
-            NumberAlignment::RightSpace => write!(writer, "{s:>width$}", width = self.width),
-            NumberAlignment::RightZero => write!(writer, "{s:0>width$}", width = self.width),
+            NumberAlignment::Left => write!(writer, "{flag}{s:<width$}", width = remaining_width),
```

Comment:
you have duplicated code here, could you please dedup it ? thanks

Reply:
done by using a `write_output` function.